app/src/main/java/com/android/ubclaunchpad/driver/session/DriverPassengersActivity.java

```diff
@@ -0,0 +1,152 @@
+
+    private static final String TAG = DriverPassengersActivity.class.getSimpleName();
+
+    private DatabaseReference session;
+    private ChildEventListener listener;
+    private DriverPassengersAdapter adapter;
+    private List<User> passengers;
+    Button startButton;
+    TextView driverName;
+    TextView driverStatus;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_driver_passengers);
+        startButton = (Button) findViewById(R.id.start_button);
+
+        String sessionName = (String) getIntent().getBundleExtra("firebaseData").get(StringUtils.FirebaseSessionName);
+        String driverUid = (String) getIntent().getBundleExtra("firebaseData").get("driverUid");
+
+        driverName = (TextView) findViewById(R.id.driver_name);
+        driverStatus = (TextView) findViewById(R.id.driver_status);
+
+        RecyclerView driversPassengers = (RecyclerView) findViewById(R.id.passengers_container);
+        passengers = new ArrayList<>();
+        adapter = new DriverPassengersAdapter(this, passengers);
+
+        driversPassengers.setAdapter(adapter);
+        driversPassengers.setLayoutManager(new LinearLayoutManager(this));
+        session = FirebaseUtils.getDatabase()
+                .child(StringUtils.FirebaseSessionEndpoint)
+                .child(sessionName);
+
+        session.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                String uid = dataSnapshot.child(StringUtils.FirebaseSessionDriverPassengers).getValue(String.class);
+                if (uid.equals(FirebaseUtils.getFirebaseUser().getUid())) {
+                    Toast.makeText(getBaseContext(), "You are the chosen one!", Toast.LENGTH_LONG).show();
+                    startButton.setVisibility(View.VISIBLE);
+                }
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        });
+
+        listener = new ChildEventListener() {
+            @Override
+            public void onChildAdded(DataSnapshot dataSnapshot, String s) {
+                addPassengerToAdapter(dataSnapshot, adapter);
+            }
+
+            @Override
+            public void onChildChanged(DataSnapshot dataSnapshot, String s) {
+            }
+
+            @Override
+            public void onChildRemoved(DataSnapshot dataSnapshot) {
+                removePassengerFromAdapter(dataSnapshot, adapter);
+            }
+
+            @Override
+            public void onChildMoved(DataSnapshot dataSnapshot, String s) {
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        };
+        session.child(StringUtils.FirebaseSessionDriverPassengers).child(driverUid).addChildEventListener(listener);
```

Comment:
this thing acts when passengers and drivers are already assigned. So basically, it just works on a list of passengers for a specific driver and updates it in the activity accordingly when it gets changed by something outside. As for passengers and drivers endpoints, Session activities are responsible for them.